File: glGo-0.0.6/src/gamedata.cpp
/*
 * gamedata.cpp
 *
 * $Id: gamedata.cpp,v 1.15 2003/10/12 00:30:50 peter Exp $
 *
 * glGo, a prototype for a 3D Goban based on wxWindows, OpenGL and SDL.
 * Copyright (c) 2003, Peter Strempel <pstrempel@gmx.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */

#ifdef __GNUG__
#pragma implementation "gamedata.h"
#endif

// For compilers that support precompilation, includes "wx.h".
#include "wx/wxprec.h"

#ifdef __BORLANDC__
#pragma hdrstop
#endif

#ifndef WX_PRECOMP
#include "wx/log.h"
#include "wx/intl.h"
#endif

#include "gamedata.h"


extern const wxString INVALID_HANDICAP_VALUE = _("Invalid handicap value: %d\nPlease select a handicap between 2 and 9.");


GameData::GameData()
{
    board_size = DEFAULT_BOARD_SIZE;
    komi = DEFAULT_KOMI;
    handicap = DEFAULT_HANDICAP;
    whiteName = _("White");
    whiteRank = wxEmptyString;
    blackName = _("Black");
    blackRank = wxEmptyString;
    result = wxEmptyString;
    gameName = wxEmptyString;
    copyright = wxEmptyString;
    place = wxEmptyString;
    date = wxEmptyString;
    time = wxEmptyString;
    filename = wxEmptyString;
    igs_type = 0;
}

GameData::GameData(const GameData &data)
{
    board_size = data.board_size;
    komi = data.komi;
    handicap = data.handicap;
    whiteName = data.whiteName;
    whiteRank = data.whiteRank;
    blackName = data.blackName;
    blackRank = data.blackRank;
    result = data.result;
    gameName = data.gameName;
    copyright = data.copyright;
    place = data.place;
    date = data.date;
    time = data.time;
    filename = data.filename;
    igs_type = data.igs_type;
}

File: ggo-0.2/src/ggo/GameData.java
/*
 *  GameData.java
 *
 *  gGo
 *  Copyright (C) 2002  Peter Strempel <pstrempel@t-online.de>
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program; if not, write to the Free Software
 *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
package ggo;

import ggo.Defines;
import ggo.gGo;
import ggo.parser.ParserDefs;
import java.io.Serializable;

/**
 *  Storage class for the data of one game. Mostly SGF fields except the
 *  filename.
 *
 *@author     Peter Strempel
 *@version    $Revision: 1.5 $, $Date: 2002/10/05 16:06:00 $
 */
public class GameData implements Defines, Serializable {
    //{{{ public members
    /**  Name of black player */
    public String playerBlack;
    /**  Name of white player */
    public String playerWhite;
    /**  Rank of black player */
    public String rankBlack;
    /**  Rank of white player */
    public String rankWhite;
    /**  Game result */
    public String result;
    /**  Date of the game */
    public String date;
    /**  Location of the game */
    public String place;
    /**  Copyright information */
    public String copyright;
    /**  Name of the game */
    public String gameName;
    /**  Filename of this game */
    public String fileName;
    /**  Board size */
    public int size;
    /**  Handicap given */
    public int handicap;
    /**  Komi given */
    public float komi;
    /**  White territory */
    public int scoreTerrWhite;
    /**  White captures */
    public int scoreCapsWhite;
    /**  White total score */
    public int scoreTerrBlack;
    /**  Black captures */
    public int scoreCapsBlack;
    /** Charset */
    public String charset;
    //}}}

    //{{{ GameData() constructor
    /**Constructor for the GameData object */
    public GameData() {
        // Init with default values
        size = 19;
        playerBlack = gGo.getBoardResources().getString("Black");
        playerWhite = gGo.getBoardResources().getString("White");
        rankBlack = "";
        rankWhite = "";
        komi = 5.5f;
        handicap = 0;
        fileName = "";
        gameName = "";
        date = "";
        place = "";
        copyright = "";
        scoreTerrWhite = scoreTerrBlack = scoreCapsWhite = scoreCapsBlack = 0;
        charset = ParserDefs.defaultgGoCharset;
    } //}}}

    //{{{ GameData(int) constructor
    /**
     *Constructor for the GameData object
     *
     *@param  boardSize  Board size
     */
    public GameData(int boardSize) {
        this();
        size = boardSize;
    } //}}}

    //{{{ toString() method
    /**
     *  Converts the GameData to a string. For debugging.
     *
     *@return    Converted string
     */
    public String toString() {
        return "[GAME DATA" +
                "\n  Black = " + playerBlack + " / " + rankBlack +
                "\n  White = " + playerWhite + " / " + rankWhite +
                "\n  Size = " + size +
                "\n  Komi = " + komi +
                "\n  Handicap = " + handicap +
                "\n  Result = " + result +
                "\n  Date = " + date + ", Place = " + place +
                "\n  Copyright = " + copyright +
                "\n  Gamename = " + gameName +
                "\n  Score white = " + scoreTerrWhite + " + " + scoreCapsWhite +
                "\n  Score black = " + scoreTerrBlack + " + " + scoreCapsBlack +
                "\n  Charset  = " + charset +
                "\n  Filename = " + fileName + "]\n";
    } //}}}
}

